torch/_dynamo/utils.py

```diff
@@ -2575,3 +2575,18 @@ class Invalid(dict):  # type: ignore[type-arg]
         pass
 
     return RemovableHandle(Invalid())
+
+
+# Returns a "proxy" (new object with the same class and dict) for (non-GraphModule) nn.Module's.
+# Attribute changes to the original object/proxy will be reflected in the other.
+# This is useful for cases where we want a keep-alive reference to a module without increasing
+# its reference count.
+def nn_module_proxy(mod):
+    if not isinstance(mod, torch.nn.Module):
+        return mod
+    if isinstance(mod, torch.fx.GraphModule):
+        # Dynamo-generated GM's shouldn't contain user-created GM's
+        return mod
+    proxy = mod.__class__.__new__(mod.__class__)
```

Comment:
hmm if module's `__new__` method expects additional arguments, maybe this won't work? e.g. per-parameter FSDP: https://github.com/pytorch/pytorch/blob/19d27a13ea052230d9fb565a5b82e683e28d1697/torch/distributed/_composable/fsdp/fully_shard.py#L142

Reply:
What kind of Modules would these be? Do we expect the dynamo-generated GraphModule to contain them?

Reply:
with inline_inbuilt_nn_modules, we won't hit this issue anymore which is nice haha